zeppelin-server/src/main/java/org/apache/zeppelin/service/JobManagerService.java

```diff
@@ -0,0 +1,157 @@
+                                          ServiceContext context,
+                                          ServiceCallback<List<NoteJobInfo>> callback)
+      throws IOException {
+    List<NoteJobInfo> notesJobInfo = new ArrayList<>();
+    Note jobNote = notebook.getNote(noteId);
+    notesJobInfo.add(new NoteJobInfo(jobNote));
+    callback.onSuccess(notesJobInfo, context);
+    return notesJobInfo;
+  }
+
+  /**
+   * Get all NoteJobInfo after lastUpdateServerUnixTime
+   */
+  public List<NoteJobInfo> getNoteJobInfoByUnixTime(long lastUpdateServerUnixTime,
+                                                    ServiceContext context,
+                                                    ServiceCallback<List<NoteJobInfo>> callback)
+      throws IOException {
+    List<Note> notes = notebook.getAllNotes();
+    List<NoteJobInfo> notesJobInfo = new ArrayList<>();
+    for (Note note : notes) {
+      NoteJobInfo noteJobInfo = new NoteJobInfo(note);
+      if (noteJobInfo.unixTimeLastRun > lastUpdateServerUnixTime) {
+        notesJobInfo.add(noteJobInfo);
+      }
+    }
+    callback.onSuccess(notesJobInfo, context);
+    return notesJobInfo;
+  }
+
+  public void removeNoteJobInfo(String noteId,
+                                ServiceContext context,
+                                ServiceCallback<List<NoteJobInfo>> callback) throws IOException {
+    List<NoteJobInfo> notesJobInfo = new ArrayList<>();
+    notesJobInfo.add(new NoteJobInfo(noteId, true));
+    callback.onSuccess(notesJobInfo, context);
+  }
+
+  private static long getUnixTimeLastRunParagraph(Paragraph paragraph) {
+    if (paragraph.isTerminated() && paragraph.getDateFinished() != null) {
+      return paragraph.getDateFinished().getTime();
+    } else if (paragraph.isRunning()) {
+      return new Date().getTime();
+    } else {
+      return paragraph.getDateCreated().getTime();
+    }
+  }
+
+
+  public static class ParagraphJobInfo {
+    private String id;
+    private String name;
+    private Job.Status status;
+
+    public ParagraphJobInfo(Paragraph p) {
+      this.id = p.getId();
+      if (StringUtils.isBlank(p.getTitle())) {
+        this.name = p.getId();
+      } else {
+        this.name = p.getTitle();
+      }
+      this.status = p.getStatus();
+    }
+  }
+
+  public static class NoteJobInfo {
+    private String noteId;
+    private String noteName;
+    private String noteType;
+    private String interpreter;
+    private boolean isRunningJob;
+    private boolean isRemoved = false;
+    private long unixTimeLastRun;
+    private List<ParagraphJobInfo> paragraphs;
+
+    public NoteJobInfo(Note note) {
+      boolean isNoteRunning = false;
+      long lastRunningUnixTime = 0;
+      this.noteId = note.getId();
+      this.noteName = note.getName();
+      // set note type ( cron or normal )
+      if (note.getConfig().containsKey("cron") && !note.getConfig()
+          .get("cron").equals("")) {
```

Comment:
consider adding a helper perhaps? should it check for get("cron") that is not whitespace only?